### PR TITLE
 C51-124: Styling drag and drop tab on files tab and other Misc issues

### PR DESCRIPTION
**Pull request by swastikpareek**
## Overview 
 Styling for Files Tab - Drag and Drop Tab on Files Tab

## After

c51-124 -- after

c51-124 - after 3

c51-124

## Technical Details
* Update Placeholder HTML.
* Adds BEM structure 
  * `civicase__file-upload-container`: Main container
  * `civicase__file-upload-col`: Columns
  * `civicase__file-upload-box`: Upload box
  * `civicase__file-upload-dropzone`: Dropzone
  * `civicase__file-upload-button`: Upload Button
  * `civicase__file-upload-details`: Details section
  * `civicase__file-upload-name`: Name 
  * `civicase__file-upload-size`: Size
  * `civicase__file-upload-progress`: Progress container 
  * `civicase__file-upload-action`: action buttons
* Adds `$upload-icon-color` for upload icon color
* UI fix for the filter dropdown on people's tab.
* Fix uploading bug when uploading after cancelling first.

### CODE REVIEW

**Comment by deb1990** [comment]
2018-09-27 at 1 42 pm

**Comment by deb1990** [comment]
@swastikpareek file upload does not work, but it works in dcase site. Please check.

File: ang/civicase/FileUploader.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+        <div class="row">
+          <div class="civicase__file-upload-name col-sm-6 col-md-8"><label>{{ts('Name')}}</label></div>
+          <div class="civicase__file-upload-size col-sm-3 col-md-2"><label>{{ts('Size')}}</label></div>
+          <div class="civicase__file-upload-action col-sm-3 col-md-2"></div>
+        </div>
+
+        <div ng-repeat="item in uploader.queue">
+          <div class="row">
+            <h4 class="civicase__file-upload-name col-sm-6 col-md-8" ng-bind="item.file.name"></h4>
```
Comment (deb1990) [inline ang/civicase/FileUploader.html]: why do we need the `md`classes?

File: ang/civicase/FileUploader.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            <div class="civicase__file-upload-size col-sm-3 col-md-2" civicase-filesize="item.file.size"></div>
+            <div class="civicase__file-upload-action col-sm-3 col-md-2 civicase__file-upload-remove">
+              <button class="btn btn-sm btn-danger-outline" ng-click="item.remove()">{{ts('Remove')}}</button>
+            </div>
+          </div>
+          <div class="progress">
```
Comment (deb1990) [inline ang/civicase/FileUploader.html]: BEM not used here and in its child elements.

File: ang/civicase/FileUploader.js [hunk 3]
```diff
@@ -46,50 +57,46 @@
...
+        case_id: $scope.ctx.id,
+        activity_type_id: 'File Upload',
+        subject: ''
+      };
+    }
+
     // TODO: Test interrupted transfer.
 
-    function pwait(delay) {
+    function pwait (delay) {
```
Comment (deb1990) [inline ang/civicase/FileUploader.js]: Whats `pwait`? please have a more meaningful name. 
Do we need the TODO? if yes, there should not be any line break.

File: ang/civicase/FileUploader.html [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            <div class="progress-bar progress-bar-success" ng-class="{fullwidth: item.progress == '100'}" role="progressbar" aria-valuenow="60" aria-valuemin="0" aria-valuemax="100" ng-style="{width: item.progress + '%'}"><span class="sr-only">{{item.progress}}% Complete</span></div>
```
Comment (deb1990) [inline ang/civicase/FileUploader.html]: I dont see the `fullwidth` in the scss. What is the usage of this?

File: ang/civicase/FileUploader.html [hunk 5]
```diff
@@ -0,0 +1,53 @@
+<form class="panel panel-default civicase__panel-transparent-header">
```
[the remark this answers is not in the document]
Comment (swastikpareek) [reply ang/civicase/FileUploader.html]: We need styles that are there for `civicase__panel-transparent-header`(same padding and box shadow) so reused this. 
So this is `civicase__panel-transparent-header` w/o `panel-header`

File: ang/civicase/FileUploader.html (hunk 2 above)
Comment (swastikpareek) [reply ang/civicase/FileUploader.html]: We din't need to add any new class here so didn't use BEM.

File: ang/civicase/FileUploader.html (hunk 1 above)
Comment (swastikpareek) [reply ang/civicase/FileUploader.html]: It was existing code and is responsible for setting up the layout. However, I feel that we can get rid of `col-sm` classes.

File: ang/civicase/FileUploader.js (hunk 3 above)
Comment (swastikpareek) [reply ang/civicase/FileUploader.js]: `pwait` was something that was coming from the previous code. So not sure what this do and therefore can't change the name.
Removed the space however.

File: ang/civicase/CaseDetailsFileTab.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  // "civicaseCaseDetailsFileTab" is a basic skeletal directive.
```
Comment (deb1990) [inline ang/civicase/CaseDetailsFileTab.js]: Unnecessary comment. please remove it.

File: ang/civicase/FileUploader.js [hunk 7]
```diff
@@ -1,43 +1,54 @@
...
-    function initActivity() {
-      $scope.activity = {
-        case_id: ctx.id,
-        activity_type_id: 'File Upload',
-        subject: ''
-      };
-    }
-    initActivity();
-    $scope.$watchCollection('ctx.id', initActivity);
+  function caseFilesController ($scope, crmApi, crmBlocker, crmStatus, FileUploader, $q, $timeout) {
```
Comment (deb1990) [inline ang/civicase/FileUploader.js]: Should be called `fileUploaderController`

**Pull request merged**